math/mathmore/src/GSLNLSMinimizer.cxx

```diff
@@ -48,23 +48,23 @@ class FitTransformFunction : public FitMethodFunction {
       fGrad( std::vector<double>(f.NDim() ) )
    {
       // constructor
-      // need to pass to MinimTransformFunction a new pointer which will be managed by the class itself
-      // pass a gradient pointer although it will not be used byb the class
+      // create a new pointer of  MinimTransformFunction, which will be managed by the class itself
    }
 
    FitTransformFunction(const FitMethodFunction & f, MinimTransformFunction *transFunc ) :
       FitMethodFunction( f.NDim(), f.NPoints() ),
-      fOwnTransformation(false),
+      fOwnTransformation(true),
```

Comment:
PR look good to me, thanks for the improvements in the ownership model!

Now that you use smart pointers, can this `fOwnTransformation` member be removed?

Reply:
Thank you for the comment. I have now improved also that internal class and the data member is removed!